cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORParser.java

```diff
@@ -1123,11 +1150,16 @@ protected JsonToken _handleTaggedBinary(TagList tags) throws IOException
             _numberBigInt = BigInteger.ZERO;
         } else {
             _streamReadConstraints.validateIntegerLength(_binaryValue.length);
-            BigInteger nr = new BigInteger(_binaryValue);
-            if (neg) {
-                nr = nr.negate();
-            }
-            _numberBigInt = nr;
+
+                BigInteger nr = new BigInteger(_binaryValue);
+                if (neg) {
+                    if (Feature.CORRECT_CBOR_NEGATIVE_BIGINT_DECODING.enabledIn(_formatFeatures)) {
+                        nr = BigInteger.ONE.negate().subtract(new BigInteger(1, _binaryValue));
```

Comment:
This throws away `nr` constructed earlier... I think I'll refactor this to remove unnecessary calculation

Reply:
makes sense, thank you